Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Merchant.hs

```diff
@@ -115,6 +127,199 @@ castMerchantStatus = \case
   DM.APPROVED -> Common.APPROVED
   DM.REJECTED -> Common.REJECTED
 
+---------------------------------------------------------------------
+merchantCommonConfigUpdate :: ShortId DM.Merchant -> Common.MerchantCommonConfigUpdateReq -> Flow APISuccess
+merchantCommonConfigUpdate merchantShortId req = do
+  runRequestValidation Common.validateMerchantCommonConfigUpdateReq req
+  merchant <- findMerchantByShortId merchantShortId
+  config <- CQTC.findByMerchantId merchant.id >>= fromMaybeM (TransporterConfigNotFound merchant.id.getId)
+  let updConfig =
+        config{pickupLocThreshold = fromMaybe config.pickupLocThreshold req.pickupLocThreshold,
+               dropLocThreshold = fromMaybe config.dropLocThreshold req.dropLocThreshold,
+               rideTimeEstimatedThreshold = fromMaybe config.rideTimeEstimatedThreshold req.rideTimeEstimatedThreshold,
+               popupDelayToAddAsPenalty = req.popupDelayToAddAsPenalty <|> config.popupDelayToAddAsPenalty, -- FIXME we can't update optional field to Nothing
```

Comment:
Maybe (Maybe Int) ???

Reply:
And how it will work on json level? Both `Nothing` and `Just Nothing` will be encoded to `null` 
I was thinking about something like this

```
driverPositionInfoExpiry :: Maybe (OptionalValue Seconds),

newtype OptionalValue a = OptionalValue {value :: Maybe a}
  deriving stock (Show, Generic)
  deriving anyclass (ToJSON, FromJSON, ToSchema)
```
on json level:
`driverPositionInfoExpiry : null` will not change anything
`driverPositionInfoExpiry : {value : null}` will reset domain value to `Nothing`
`driverPositionInfoExpiry : {value : a}` will set domain value to `(Just a)`

But I'm not sure this is required

Reply:
Oh, you are right. I just remember we had some update code like this somewhere, seems like it never worked, huh? Can't find it now though. Then i think you should create that newtype, mb in another PR though, coz you'll need to change all maybes in requests to this type